### PR TITLE
Implement advance login

### PR DESCRIPTION
**Pull request by NguyenMinhTuan2906**

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -3,16 +3,41 @@ def log_in user
...
+    end
+  end
+
   def logged_in?
     current_user.present?
   end
 
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```
Comment (ghost) [inline app/helpers/sessions_helper.rb]: remove () ?

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -3,16 +3,41 @@ def log_in user
...
+    cookies.delete(:remember_token)
```
Comment (ghost) [inline app/helpers/sessions_helper.rb]: as above

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -3,16 +3,41 @@ def log_in user
...
+  end
+
   def log_out
-    session.delete :user_id
+    forget(current_user)
+    session.delete(:user_id)
```
Comment (ghost) [inline app/helpers/sessions_helper.rb]: as above

File: app/models/user.rb [hunk 4]
```diff
@@ -17,4 +18,22 @@ def User.digest string
...
     BCrypt::Password.create(string, cost: cost)
   end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```
Comment (ghost) [inline app/models/user.rb]: update_attributes

File: app/models/user.rb [hunk 5]
```diff
@@ -17,4 +18,22 @@ def User.digest string
...
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (ghost) [inline app/models/user.rb]: Dùng update_attributes

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -152,6 +152,20 @@ aside {
 
 /* forms */
 
+.checkbox {
+  margin-top: -10px;
+  margin-bottom: 10px;
+  span {
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss]: space while bettwen block

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -152,6 +152,20 @@ aside {
 
 /* forms */
 
+.checkbox {
+  margin-top: -10px;
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss]: order by name
margin-bottom:
margin-top:

File: app/assets/stylesheets/custom.scss [hunk 8]
```diff
@@ -152,6 +152,20 @@ aside {
...
+  margin-bottom: 10px;
+  span {
+    font-weight: normal;
+    margin-left: 20px;
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss]: order by name

File: app/assets/stylesheets/custom.scss [hunk 9]
```diff
@@ -152,6 +152,20 @@ aside {
...
+  }
+}
+
+#session_remember_me {
```
Comment (ghost) [inline app/assets/stylesheets/custom.scss:165]: Hạn chế dùng id selector trong css

File: app/models/user.rb [hunk 10]
```diff
@@ -17,4 +18,22 @@ def User.digest string
...
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attributes remember_digest: User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
```
Comment (hungnh103) [inline app/models/user.rb]: remove `()`

**Pull request merged**